r/R/dplyr.R

```diff
@@ -364,6 +383,7 @@ set_filters <- function(.data, expressions) {
 
 collect.arrow_dplyr_query <- function(x, as_data_frame = TRUE, ...) {
   x <- ensure_group_vars(x)
+  x <- ensure_arrange_vars(x) # this sets x$temp_columns
```

Comment:
Does `temp_columns` need to be added to `x`? It looks like it doesn't live outside of this function. You could spell this like

```
temp_columns <- find_extra_arrange_columns(x)
```

Reply:
I would also add a comment that explains how, unlike group_by_vars, which always are kept even if you don't `select` them, we have to do this for sorting because you could arrange then select without them but because of how the lazy query is built, we don't carry out the sort until later.

Reply:
Including `temp_columns` in `x` makes it possible to pass it to `Scanner$create()` without changing its interface when we want to support `arrange()` on Datasets.

Reply:
Lengthy comment added in d53abf885fe33afbf1079dbae439e1753efd071e